src/Microsoft.AspNet.Razor.Runtime/TagHelpers/HtmlAttributeNameAttribute.cs

```diff
@@ -29,5 +31,39 @@ public HtmlAttributeNameAttribute(string name)
         /// HTML attribute name of the associated property.
         /// </summary>
         public string Name { get; }
+
+        /// <summary>
+        /// Gets or sets the prefix used to match HTML attribute names. Matching attributes are added to the
+        /// associated property (an <see cref="System.Collections.Generic.IDictionary{TKey, TValue}"/>).
+        /// </summary>
+        /// <remarks>
+        /// If non-<c>null</c> associated property must be compatible with
+        /// <see cref="System.Collections.Generic.IDictionary{TKey, TValue}"/> where <c>TKey</c> is
+        /// <see cref="string"/>.
+        /// </remarks>
+        /// <value>
+        /// If associated property is compatible with
+        /// <see cref="System.Collections.Generic.IDictionary{TKey, TValue}"/>, default value is <c>Name + "-"</c>.
+        /// Otherwise default value is <c>null</c>.
+        /// </value>
+        public string DictionaryAttributePrefix
+        {
+            get
+            {
+                return _dictionaryAttributePrefix;
+            }
+            set
+            {
+                _dictionaryAttributePrefix = value;
+                DictionaryAttributePrefixSet = true;
+            }
+        }
+
+        /// <summary>
+        /// Gets an indication whether <see cref="DictionaryAttributePrefix"/> has been set. Used to distinguish an
+        /// uninitialized <see cref="DictionaryAttributePrefix"/> value from an explicit <c>null</c> setting.
```

Comment:
We have discussed this one to death and your approach is just weird.  To remind you:
- Dictionary properties are _always_ identical to any other property and may be set in the Razor source if the author chooses.  (System does not support read-only dictionary properties.)
- Prefix matching is optional but on by default for dictionary properties.
- To disable prefix matching, use `[HtmlAttributeName("someName", DictionaryAttributePrefix = null)]`
- To override the default prefix, use `[HtmlAttributeName("someName", DictionaryAttributePrefix = "prefix")]`
- There is never an ambiguity between attribute names and prefixes even in the case of `[HtmlAttributeName("someName", DictionaryAttributePrefix = "someName")]` because non-prefix matches have precedence.